### PR TITLE
Refactor subsystem code

### PR DESCRIPTION
**Pull request by lasote**
Changelog: Feature: New `conan.tool.microsoft.unix_path` to convert paths to any subsystem when using `conanfile.win_bash`.
Changelog: Fix: Removed `subsystem_path` from the `conan.tool.microsoft` namespace, superseded by `unix_path`.
Docs: https://github.com/conan-io/docs/pull/2479

### CODE REVIEW

File: conans/client/subsystems.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
+import os
```
Comment (lasote) [inline conans/client/subsystems.py:1]: All this moved has been just moved. Not a single line change.

File: conan/tools/microsoft/subsystems.py [hunk 2]
```diff
@@ -1,174 +1,8 @@
...
-                                 "needed to run commands in a Windows subsystem")
-    return subsystem
-
-
-def subsystem_path(subsystem, path):
-    """"Used to translate windows paths to MSYS unix paths like
-    c/users/path/to/file. Not working in a regular console or MinGW!
-    """
-    if subsystem is None or subsystem == WINDOWS:
+def unix_path(conanfile, path, subsystem=None):
```
Comment (memsharded) [inline conan/tools/microsoft/subsystems.py]: I don't see the case for hardcoding the subsystem in recipes, that probably shouldn't be added atm.

**Review by memsharded — CHANGES_REQUESTED** [review]

File: conan/tools/microsoft/subsystems.py (hunk 2 above)
Comment (lasote) [reply conan/tools/microsoft/subsystems.py:4]: Wow, I don't know what happened with the rest of the code 😅 .
It is fixed and the test completed.

**Comment by lasote** [comment]
Review again please

**Review by memsharded — APPROVED** [review]

**Pull request merged**